### PR TITLE
Add ignore_cert option to ignore TLS/SSL certificate errors for self sig...

### PR DESCRIPTION
**Pull request by thet**
...ned certificates when using the ldaps uri schema.

### CODE REVIEW

**Comment by thet** [comment]
@rnixx @jensens @saily FYI

**Comment by jensens** [comment]
Tests are red?

File: src/node/ext/ldap/session.py [hunk 1]
```diff
@@ -76,10 +76,15 @@ def authenticate(self, dn, pw):
...
         """
         # Let's bypass connector/communicator until they are sorted out
+        if self._props.ignore_cert:
+            ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_NEVER)
         con = ldap.initialize(self._props.uri)
         try:
             con.simple_bind_s(dn, pw)
-        except ldap.INVALID_CREDENTIALS:
+        except (ldap.INVALID_CREDENTIALS, ldap.UNWILLING_TO_PERFORM) as e:  # noqa
+            # unwilling to perform can happen, if you quert a local user named
```
Comment (rnixx) [inline src/node/ext/ldap/session.py]: typo -> query

**Comment by rnixx** [comment]
tests red since https://travis-ci.org/bluedynamics/node.ext.ldap/builds/49952225

what happened to the test data?

**Comment by thet** [comment]
i can reproduce the test failures locally on my machine. but i do not understand, what's happening. looks pretty much like a side effect.

**Comment by rnixx** [comment]
tests are green again (at least on my local machine). There was a side effect between README tests and principal tests. please adopt your pull request

**Comment by thet** [comment]
just rebased and updated this pull-request. waiting for travis.

**Comment by rnixx** [comment]
https://travis-ci.org/bluedynamics/node.ext.ldap/builds/50621969

what's going on here?

**Comment by jensens** [comment]
Here an upgrade step is missing adding the new property to existing instances!

**Pull request merged**